src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1405,27 +1658,22 @@ internal int Count
         /// </param>
         internal void Start(Process process, NativeCommandIOFormat inputFormat)
         {
-            Dbg.Assert(process != null, "caller should validate the parameter");
+            Dbg.Assert(process != null, "caller should validate the paramter");
 
             //Get the encoding for writing to native command. Note we get the Encoding 
             //from the current scope so a script or function can use a different encoding
             //than global value.
             Encoding pipeEncoding = _command.Context.GetVariableValue(SpecialVariables.OutputEncodingVarPath) as System.Text.Encoding ??
                                     Encoding.ASCII;
 
-            _streamWriter = new StreamWriter(process.StandardInput.BaseStream,
-                                            pipeEncoding);
-            _inputFormat = inputFormat;
+            _streamWriter = new StreamWriter(process.StandardInput.BaseStream, pipeEncoding);
+            _streamWriter.AutoFlush = true;
 
-            if (inputFormat == NativeCommandIOFormat.Text)
-            {
-                ConvertToString();
-            }
-            _inputThread = new Thread(new ThreadStart(this.WriterThreadProc));
-            _inputThread.Start();
+            _inputFormat = inputFormat;
```

Comment:
`_inputFormat` was used for processing `Text` or `Xml` input. If it's proven that we can ignore `Xml` input, then we need to clean up `ProcessInputWriter` to remove this member, including the parameter of `Start` method.